### PR TITLE
add setOptions in LGeoJson. Update example

### PR DESCRIPTION
**Pull request by bezany**
Recreate pull-request [#226](https://github.com/KoRiGaN/Vue2Leaflet/pull/226) (create branch)

### CODE REVIEW

**Comment by bezany** [comment]
I think use setStyle function very good for performance. But then need compare old and new options objects for detect that changed: only style property or not. If changed only style property, then use set Style. Otherwise create layer.
For compare two object need util function like [this](https://stackoverflow.com/a/1144249)
Later I can try to implement the required functionality.

**Comment by lordfuoco** [comment]
@bezany my point is, do we really need the rest of the options to be reactive ? 
since the use case to me is related to the geoJSON style.
the rest of the options can still be accessed through the ref, since they are all functions.

**Comment by lordfuoco** [comment]
@bezany merged the rest, can we finish the discussion about setStyle ? 
do we really need to have the full options reactive ? can you present me an use case ?

**Comment by lordfuoco** [comment]
@bezany I think we can refactor you work to suit your use case ( which I think is the common use case 'even for me!' ) keep options as `custom:false` and add a separate `style` prop with `custom:true` and call the set style function.

What do you think ?

**Comment by bezany** [comment]
@lordfuoco I have the same idea that you have - create `custom:true` property for style (name `style` reserved in Vue,  I propose use `optionsStyle`). And if the `optionsStyle` changes then use `mapObject.setStyle` (use `mapObject.eachLayer` not neccesary, [leafelt uses `eachLayer` in `setStyle`](https://github.com/Leaflet/Leaflet/blob/ba1b253126df92b91993a678050d18a99d3d7b63/src/layer/GeoJSON.js#L141))

About `options` keep `custom:false`. In this case change `options` nothing change on the map. I see that Leafelt not re-render layer. May be for 'vue-way' set `custom:true` and use `clearLayers-addData` for real re-render?

But how merge `oprtions.style` and `optionsStyle`? I propose use optionsStyle if it set. And use `options.style` otherwise.

I'll soon finish the commit.

**Comment by bezany** [comment]
I done example with `options` and `optionsStyle`. In example dynamically changes `onEachFeature` in `options` property for show/hide tooltip. For change color used `oprtionsStyle` that use fast method `mapObject.setStyle`.
example-geojson-change-style

File: src/components/LGeoJson.vue [hunk 1]
```diff
@@ -32,6 +38,17 @@ export default {
   beforeDestroy () {
     this.parentContainer.mapObject.removeLayer(this.mapObject);
   },
+  computed: {
+    mergedOptions () {
+      if (this.optionsStyle) {
+        const options = L.Util.extend({}, this.options)
```
Comment (lordfuoco) [inline src/components/LGeoJson.vue]: I think we should prefer ES6 syntax to do this, the whole block can be change with 
```
{
...this.options,
style: this.optionsStyle ? this.optionsStyle : this.options.style
}

File: src/components/LGeoJson.vue [hunk 2]
```diff
@@ -10,8 +11,13 @@ const props = {
   },
   options: {
     type: Object,
+    custom: true,
     default: () => ({})
   },
+  optionsStyle: {
+    type: [Object, Function],
+    custom: true
```
Comment (lordfuoco) [inline src/components/LGeoJson.vue]: Please add a `default: null`

**Comment by lordfuoco** [comment]
@bezany since you are working on the codebase so much maybe is worth that you take a look at this:
https://github.com/KoRiGaN/Vue2Leaflet/pull/231

**Comment by lordfuoco** [comment]
@bezany great work! Tomorrow I am going to merge this.

File: src/components/LGeoJson.vue (hunk 2 above)
Comment (bezany) [reply src/components/LGeoJson.vue]: done!

File: src/components/LGeoJson.vue (hunk 1 above)
Comment (bezany) [reply src/components/LGeoJson.vue]: done! I use operator `||`.

**Comment by bezany** [comment]
Perfectly!
I'll look at the #231 in the presence of time.

**Pull request merged**